pydatalab/pydatalab/apps/chat/blocks.py

```diff
@@ -14,23 +14,7 @@
 MODEL = "gpt-3.5-turbo-0613"
 MAX_CONTEXT_SIZE = 4097
 
-
-def num_tokens_from_messages(messages: Sequence[dict]):
-    # see: https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb
-    encoding = tiktoken.encoding_for_model(MODEL)
-
-    tokens_per_message = 4  # every message follows <|start|>{role/name}\n{content}<|end|>\n
-    tokens_per_name = -1  # if there's a name, the role is omitted
-
-    num_tokens = 0
-    for message in messages:
-        num_tokens += tokens_per_message
-        for key, value in message.items():
-            num_tokens += len(encoding.encode(value))
-            if key == "name":
-                num_tokens += tokens_per_name
-    num_tokens += 3  # every reply is primed with <|start|>assistant<|message|>
-    return num_tokens
+openai_client = ChatOpenAI(api_key=os.environ.get("OPENAI_API_KEY"), model=MODEL)
```

Comment:
The tests are currently failing on other blocks because this client fails to initialize without an API key. Probably the creation of this client should be pushed inside a block (and made separately for each new block), as e.g.,

```python
def __init__(self, ...):
    self.openai_client = ChatOpenAI(api_key=os.environ.get("OPENAI_API_KEY"), model=MODEL)

```

Reply:
I have fixed this now.